### ISSUE TITLE
Error at line 57, file /home/username/gddr6/lib/src/gddr6.c (1) [Operation not permitted]

### ISSUE THREAD

**Issue by Ahmed-Mahmod-Salem** [olealgoritme/gddr6#29]
the code fails at this point everytime I try to run it

**Comment by olealgoritme** [comment]
Did you enable iomem=relaxed and run it with sudo?

**Comment by Ahmed-Mahmod-Salem** [comment]
yes, I did everything in the description, didn't work, turns out it is a problem because we are trying to access /dev/mem which is locked by the kernel, had to recompile the kernel, and disabled secure boot, don't really know for sure which one of the two attempts did it, maybe it was both, but finally it worked.

**Comment by daniel-dona** [comment]
Getting the same error when running the gddr6 binary:

```
$ sudo gddr6 
Error at line 57, file /home/dani/gddr6/lib/src/gddr6.c (1) [Operation not permitted]
```

I verified that iomem (`iomem=relaxed`) parameter is in the boot command line, kernel is 5.19.0-41-generic (Ubuntu 22.04)

I can provide additional information of the system if needed

**Comment by Ahmed-Mahmod-Salem** [comment]
yeah, it is a deep hassle, my advice, just don't.

**Comment by daniel-dona** [comment]
Additional info from the kernel:

```
[ 8903.861420] Lockdown: gddr6: /dev/mem,kmem,port is restricted; see man kernel_lockdown.7
```
Looks like is related to UEFI secure boot
```
$ sudo mokutil --sb
SecureBoot enabled
```
I'll test to disable secure boot and see if it works

**Comment by daniel-dona** [comment]
It was that!

```
$ sudo gddr6
Device: RTX 3080 Ti GDDR6X (GA102 / 0x2208) pci=2d:0:0
VRAM Temps: |  52°C |
```

**Comment by daniel-dona** [comment]
> yeah, it is a deep hassle, my advice, just don't.

With this Ubuntu kernel just the iomem=relaxed and secure boot disabled works, no kernel recompile required. Maybe this can be added in the README, I can create a PR @olealgoritme if you agree...

**Comment by olealgoritme** [comment]
@daniel-dona Yes, please create a PR that highlights necessary UEFI Secure Boot setting in the README. Thank you!
